app/jobs/update_jobs/adjust_question_reflections.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class UpdateJobs::AdjustQuestionReflections < CloneJob
+  def perform(question, prev_variable_name)
+    reflectable_questions(question).each do |q|
+      q.narrator['blocks'].each do |block|
+        next if block['reflections'].nil?
+
+        block['reflections'].each do |reflection|
+          next unless reflection['variable'] == prev_variable_name
+
+          reflection['variable'] = question.body['variable']['name']
+        end
+
+        q.save!
+      end
+    end
+  end
+
+  def reflectable_questions(question)
+    Question
+       .within_intervention(question.session.intervention_id)
+       .where.not(id: question.id)
+       .joins("CROSS JOIN jsonb_array_elements(narrator->'blocks') AS narrator_block")
```

Comment:
I'm wondering if `CROSS JOIN` is necessary here.

Reply:
Unfortunatelly CROSS JOIN is necessary, because with outher joins this doesn't work